### PR TITLE
feat: Bookmark

### PR DESCRIPTION
**Pull request by Vishruta-Patil**
- Get Bookmark Posts
- Add bookmark
- Remove from bookmark

### CODE REVIEW

File: src/components/common/modal/EditPost.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { useAppSelector } from "hooks";
+import { Avatar } from "../avatar/Avatar";
+import { HeroBtn } from "../button/HeroBtn";
+
+export const EditPostModal = () => {
+  const { avatar } = useAppSelector((store) => store.auth);
+  return (
+    // fixed p-4 left-0 right-0 top-0 h-screen z-50 flex justify-center items-center bg-overlay-color
```
Comment (vermasavita) [inline src/components/common/modal/EditPost.tsx]: remove these comments

File: src/services/authService.ts [hunk 2]
```diff
@@ -166,21 +159,96 @@ export const getAvatarProfile = (createAsyncThunk as any)(
...
+// console.log("asd" + returnPromise)
+
+// const res1 = response.items.reduce(async(acc:any, curr) => {    // (in promise) TypeError: acc.push is not a function
+//   const reqURL = getDownloadURL(curr)
+//   let obj:any= {}
+//   obj[curr.name] = reqURL
+//   acc.push(obj)
+//   return acc
+// }, [])
+// // console.log({res1})
```
Comment (vermasavita) [inline src/services/authService.ts]: You can remove these comments if it will not be used

**Review by vermasavita — APPROVED** [review]
Just a small suggestion that you can change the avatar on onChange event.

File: src/services/authService.ts [hunk 3]
```diff
@@ -166,21 +157,55 @@ export const getAvatarProfile = (createAsyncThunk as any)(
...
 
-      if(profileExists) {
-      reqURL = (async () => await getDownloadURL(getProfile))() 
-      } else reqURL= "no_image"
-      
-     return reqURL
-      
+      return reqURL;
     } catch (err) {
       console.log(err);
```
Comment (sri-rishi) [inline src/services/authService.ts:175]: suggestion: for errors you can use `console.error`

**Review by sri-rishi — APPROVED** [review]
code looks good and functionality is also working fine , feel free to merge

**Pull request merged**